src/dotnet/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -3,59 +3,78 @@
+    /// <param name="props"></param>
+    /// <returns>DateTimeOffset representing end period of carbon aware data search. </returns>
+    /// <exception cref="ArgumentException">Throws exception if prop isn't a valid DateTimeOffset. </exception>
+    private DateTimeOffset GetOffsetOrDefault(IDictionary props, string field, DateTimeOffset defaultValue) 
+    {
+        // Default if null
+        var dateTimeOffset = props[field] ?? defaultValue;
 
+        // If fail to parse property, throw error
+        if (!DateTimeOffset.TryParse(dateTimeOffset.ToString(), out defaultValue))
+        {
+            Exception ex = new ArgumentException("Failed to parse" + field + "field. Must be a valid DateTimeOffset");
+            _logger.LogError("argument exception", ex);
+            throw ex;
         }
 
-        /// <summary>
-        /// Extracts the given offset prop and converts to DateTimeOffset. If prop is not defined, defaults to provided default
-        /// </summary>
-        /// <param name="props"></param>
-        /// <returns>DateTimeOffset representing end period of carbon aware data search. </returns>
-        /// <exception cref="ArgumentException">Throws exception if prop isn't a valid DateTimeOffset. </exception>
-        private DateTimeOffset GetOffsetOrDefault(IDictionary props, string field, DateTimeOffset defaultValue) 
+        return defaultValue;
+    }
+
+    private IEnumerable<Location> GetLocationOrThrow(IDictionary props) {
+        if (props[CarbonAwareConstants.Locations] is IEnumerable<Location> locations)
         {
-            // Default if null
-            var dateTimeOffset = props[field] ?? defaultValue;
-
-            // If fail to parse property, throw error
-            if (!DateTimeOffset.TryParse(dateTimeOffset.ToString(), out defaultValue))
-            {
-                Exception ex = new ArgumentException("Failed to parse" + field + "field. Must be a valid DateTimeOffset");
-                _logger.LogError("argument exception", ex);
-                throw ex;
-            }
-
-            return defaultValue;
+            return locations;
         }
+        Exception ex = new ArgumentException("locations parameter must be provided and be non empty");
+        _logger.LogError("argument exception", ex);
+        throw ex;
+    }
 
-        private IEnumerable<Location> GetLocationOrThrow(IDictionary props) {
-            if (props[CarbonAwareConstants.Locations] is IEnumerable<Location> locations)
-            {
-                return locations;
-            }
-            Exception ex = new ArgumentException("locations parameter must be provided and be non empty");
-            _logger.LogError("argument exception", ex);
-            throw ex;
+    public async Task<double> CalcEmissionsAverageAsync(IDictionary props)
+    {
+        ValidateAverageProps(props);
+        var list = await GetEmissionsDataAsync(props);
+        var value = list.Any() ? list.Select(x => x.Rating).Average() : 0;
```

Comment:
Optional: Super readable but might be worth a comment explaining what you're doing here due to use of syntactic sugar.

I also respect the idea that code is self-documenting and so if you think this is excessive, I'm for that as well

Reply:
Sure, I'll add a comment. Thx